snapcraft/tests/test_commands_enable_ci.py

```diff
@@ -0,0 +1,104 @@
+
+    def test_enable_ci_empty(self):
+        with self.assertRaises(SystemExit) as raised:
+            main(['enable-ci'])
+
+        self.assertEqual(1, raised.exception.code)
+        self.assertEqual([
+            'Please select one of the supported integration systems: travis.'
+        ], self.fake_logger.output.splitlines())
+
+    def test_enable_ci_unknown(self):
+        with self.assertRaises(SystemExit) as raised:
+            main(['enable-ci', 'bazinga'])
+
+        self.assertEqual(1, raised.exception.code)
+        self.assertEqual([
+            '"bazinga" integration is not supported by snapcraft.',
+            'Please select one of the supported integration systems: travis.'
+        ], self.fake_logger.output.splitlines())
+
+    @mock.patch.object(travis, 'enable')
+    @mock.patch('builtins.input')
+    def test_enable_ci_travis(self, mock_input, mock_enable):
+        mock_input.side_effect = ['y']
+        main(['enable-ci', 'travis'])
+
+        self.assertEqual(1, mock_enable.call_count)
+        self.assertEqual([
+            'Snapcraft integration for Travis (CI).',
+            '',
+            'This command currently depends on working `travis` CLI '
+            'environment and',
+            'a previously initialised Travis project (`.travis.yml`).',
+            '',
+            'Make sure your Travis project is also configured to '
+            '"Build pushes", this',
+            'way every new push to `master` will result in a new snap '
+            'revision on the',
+            'Store.',
+            '',
+            'This operation will acquire properly attenuated Store '
+            'credentials and',
+            'encrypt it for use in your testbed (`.travis_snapcraft.cfg`), '
+            'only Travis',
+            'has the private key to decrypt it and will be only available '
+            'to branches',
+            'of the same repository, not forks.',
+            '',
+            "Then it will adjust Travis configuration ('.travis.yml') with "
+            'the commands',
+            'to decrypt credentials and install and run `snapcraft` to '
+            'release your snap',
+            '(inside a ubuntu:xenial docker container) during the '
+            "'after_success' phase.",
+            'See the example below::',
+            '',
+            '    sudo: required',
+            '    services:',
+            '    - docker',
+            '    after_success:',
+            '    - openssl aes-256-cbc -K <travis-key> -iv <travis-iv>',
+            '      -in .travis_snapcraft.cfg -out .snapcraft.cfg -d',
+            '    - if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then',
+            '      docker run -v $(pwd):$(pwd) -t ubuntu:xenial',
+            '      sh -c "apt update -qq && apt install snapcraft -y && '
+            'cd $(pwd) &&',
+            '      snapcraft && snapcraft push *.snap --release edge";',
+            '      fi',
+            ''
```

Comment:
It's nice being able to test that this works as expected, but I'm afraid that duplicating the large text here in its entirety will end up being difficult to maintain (i.e. fix a typo, have to change the test). Would the test be as robust if we used some regex to just make sure the right thing is printed, for example?